App/Components/Common/DropDown/DropDown.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import { View, ViewPropTypes, StyleSheet, Picker } from 'react-native';
+import PropTypes from 'prop-types';
+import Lodash from 'lodash';
+
+// Theme
+import { fonts, colors, toBaseDesignPx, spacers } from '../../../Core/Theme';
+
+// Common
+import Text from '../Text';
+
+class DropDownComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: null,
+    };
+  }
+
+  handleOptionChange = value => {
+    const { onChange } = this.props;
+    this.setState({ value });
+    onChange(value);
+  };
+
+  renderLabel = () => {
+    const { label, labelStyle, hasError } = this.props;
+    const errorLabelStyle = hasError ? styles.errorLabelStyle : null;
+
+    if (label) {
+      return (
+        <Text.SemiBold text={label} style={[this.getLabelStyle(), errorLabelStyle, labelStyle]} />
+      );
+    }
+
+    return null;
+  };
+
+  renderOptions = () => {
+    const { options } = this.props;
+    if (!options) return null;
+    return options.map(obj => <Picker.Item key={obj.value} label={obj.label} value={obj.value} />);
```

Comment:
estas creando una funcion anonima en cada render ahi, pon el callback en una funcion: ejemplo: options.map(this.renderPickerItems) y la funcion recibe como su parametro el objItem